projecttacoma/fqm-testify
# Cqfm test case Measure Report

# Summary
Added functionality for generating cqfm-test-case MeasureReports for each test case to be exported as part of the test case bundle

## New Behavior
when exporting a test case, there should now be an additional MeasureReport resource which is a cqfm-test-case MeasureReport

## Code Changes
- added functionality for creating the test case measure report in resourceCreation.ts
- appropriately updated all places where information that should cause the cqfm-test-case MeasureReport to update
- added testing

# Testing Guidance
- `npm run check`
- `npm run dev`
- upload a measure
- create a patient
- export the patient and ensure that a cqfm-test-case MeasureReport exists in the exported bundle
- import a patient without a cqfm-test-case MeasureReport in its bundle
- export the imported patient and ensure that a cqfm-test-case MeasureReport is present in the bundle
- import multiple patients, then export and ensure that a cqfm-test-case MeasureReport is present in each bundle
- return to the measure upload page and change the measure and measurement period
- return to the reseource creation page and export all resources. Note that the cqfm-test-case MeasureReport will now reflect data from the new measure.
- Link to [cqfm-test-case MeasureReport docs](https://build.fhir.org/ig/HL7/cqf-measures/StructureDefinition-test-case-cqfm.html)
- Ensure all present data are as specified in the docs
